src/api/middleware/controllers/book.js

```diff
@@ -0,0 +1,88 @@
+const Book = require("../../../models/Book");
+const { isValidObjectId } = require("../../../utils.js/validator");
+
+module.exports = {
+  create: async (req, res) => {
+    const {
+      title,
+      description,
+      subject,
+      total_quantity,
+      dimension,
+      total_price,
+      authorInformation,
+    } = req.body;
+
+    if (!isValidObjectId(authorInformation))
+      return res.json({ message: "authorInformation is invalid " });
+    if (!title) return res.json({ message: "title is required" });
+    if (!description) return res.json({ message: "description is required" });
+    if (!subject) return res.json({ message: "subject is required" });
+    if (!total_quantity)
+      return res.json({ message: "total_quantity is required" });
+    if (!dimension) return res.json({ message: "dimension is required" });
+    if (!total_price) return res.json({ message: "total_price is required" });
+
+    let quantity = { inStock: total_quantity, rentedout: 0 };
+    let price = { dailyRental: total_price, currency: "#" };
+    let book = new Book({
+      title,
+      description,
+      subject,
+      price,
+      quantity,
+      dimension,
+      authorInformation,
+    });
+
+    book = await book.save();
+    res.json(book);
+  },
+
+  //update
+  update: async (req, res) => {
+    const { id } = req.params;
+    const { subject } = req.body;
+    if (!isValidObjectId(id))
+      return res.json({ message: "Book Id is invalid" });
+    if (!subject) return res.json({ message: "subject is required" });
+
+    let book = await Book.findById(id).exec();
+    if (!book) return res.status(400).json({ message: "Book not found" });
+    book.subject = subject;
+
```

Comment:
You are updating only the book subject, you need to update the other information like title, description, price, quantity, dimension as well.